libtransact/src/scheduler/serial/execution.rs

```diff
@@ -42,23 +47,41 @@ impl Iterator for SerialExecutionTaskIterator {
 
     /// Return the next execution task which is available to be executed.
     fn next(&mut self) -> Option<ExecutionTask> {
+        if self.is_complete {
+            error!(
```

Comment:
This should be debug - I don't think this is technically an error condition, as we're not putting the iterator into a state that is invalid (i.e. sending a message for next to the scheduler core when we know there are no more tasks).

Reply:
Updated to `debug!`